packages/flakybot/src/config.ts

```diff
@@ -0,0 +1,183 @@
+    if (parsed.ext === '.yml') {
+      this.badConfigPaths.push(`${parsed.dir}/${parsed.name}.yaml`);
+    } else if (parsed.ext === '.yaml') {
+      this.badConfigPaths.push(`${parsed.dir}/${parsed.name}.yml`);
+    }
+    this.configName = parsed.name;
+  }
+  private async validateConfig(
+    configYaml: string
+  ): Promise<ValidateConfigResult> {
+    const validateSchema = this.ajv.compile(this.schema);
+    const config = (yaml.load(configYaml) as unknown) as T;
+    let isValid = false;
+    let errorText: string | undefined;
+    if (typeof config === 'object') {
+      isValid = validateSchema(config);
+      if (!isValid) {
+        errorText = JSON.stringify(validateSchema.errors, null, 4);
+      }
+    } else {
+      errorText = `.github/${this.configPath} is not valid YAML 😱`;
+    }
+    return {isValid: isValid, errorText: errorText};
+  }
+
+  public async validateConfigChanges(context: Context): Promise<void> {
+    const listFilesParams = context.repo({
+      pull_number: context.payload.pull_request.number,
+      per_page: 100,
+    });
+    const pullRequestCommitSha = context.payload.pull_request.head.sha;
+    try {
+      let errorText = '';
+      const files = await context.octokit.paginate(
+        context.octokit.pulls.listFiles,
+        listFilesParams
+      );
+      for (let i = 0; files[i] !== undefined; i++) {
+        const file = files[i];
+        if (file.status === 'removed') {
+          continue;
+        }
+        logger.debug(`file: ${file.filename}`);
+        if (this.badConfigPaths.indexOf(file.filename) > -1) {
+          // Trying to add a config file with a wrong file extension.
+          errorText +=
+            `You tried to add ${file.filename}, ` +
+            `but the config file must be ${this.configPath}\n`;
+        }
+        if (file.filename === this.configPath) {
+          const blob = await context.octokit.git.getBlob(
+            context.repo({
+              file_sha: file.sha,
+            })
+          );
+          const fileContents = Buffer.from(
+            blob.data.content,
+            'base64'
+          ).toString('utf8');
+          const result = await this.validateConfig(fileContents);
+          if (!result.isValid) {
+            errorText += result.errorText;
+          }
+        }
+        if (errorText !== '') {
+          const checkParams = context.repo({
+            name: `${this.configName} config schema`,
+            conclusion: 'failure' as Conclusion,
+            head_sha: pullRequestCommitSha,
+            output: {
+              title: 'Config schema error',
+              summary: 'An error found in the config file',
+              text: errorText,
+            },
+          });
+          await context.octokit.checks.create(checkParams);
+        }
+      }
+    } catch (err) {
+      logger.error(err);
+      return;
+    }
+  }
+}
+
+export async function getConfig<T>(
+  context: OctokitContext,
```

Comment:
This one could just provide the octokit instance, rather than a probot Context.

Reply:
Will do